src/components/Main/index.tsx

```diff
@@ -1,13 +1,22 @@
-import React from 'react';
-import { Navigate, Route, Routes } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { Navigate, Route, Routes, useLocation } from 'react-router-dom';
 import styled from '@emotion/styled';
 import { redirects, routes } from '@routes';
 import Header from '@components/Header';
 import useCurrentPage from '@hooks/useCurrentPage';
+import { useScrollToTop } from '@hooks/useScrollToTop';
 import { ANGOLA_STYLES } from '@styles/commonStyles';
+import ToTopButton from './ToTopButton';
 
 const Main = () => {
   const { title, name, params, search } = useCurrentPage();
+  const location = useLocation();
```

Comment:
```suggestion
  const { title, name, params, search, location } = useCurrentPage();
```

`useCurrentPage` 훅 내에서도 `useLocation`을 사용해 `location`을 사용하는 부분이 있어서, `useCurrentPage`에서 이 `location`을 반환해 사용하는 것도 좋을 것 같아요 _!_ _!_ 
```
const useCurrentPage = (): CurrentPage => {
 const location = useLocation(); 
```